javascript/src/game.js

```diff
@@ -1,171 +1,173 @@
-const Game = function () {
-  var players          = new Array();
-  var places           = new Array(6);
-  var purses           = new Array(6);
-  var inPenaltyBox     = new Array(6);
+const Game = function() {
+  this.logs = [];
+  this.log = text => {
+    console.log(text);
+    this.logs.push(text);
+  };
+
+  var players = new Array();
+  var places = new Array(6);
+  var purses = new Array(6);
+  var inPenaltyBox = new Array(6);
 
-  var popQuestions     = new Array();
+  var popQuestions = new Array();
   var scienceQuestions = new Array();
-  var sportsQuestions  = new Array();
-  var rockQuestions    = new Array();
+  var sportsQuestions = new Array();
+  var rockQuestions = new Array();
 
-  var currentPlayer    = 0;
+  var currentPlayer = 0;
   var isGettingOutOfPenaltyBox = false;
 
-  var didPlayerWin = function(){
-    return !(purses[currentPlayer] == 6)
+  var didPlayerWin = () => {
+    return !(purses[currentPlayer] === 6);
   };
 
-  var currentCategory = function(){
-    if(places[currentPlayer] == 0)
-      return 'Pop';
-    if(places[currentPlayer] == 4)
-      return 'Pop';
-    if(places[currentPlayer] == 8)
-      return 'Pop';
-    if(places[currentPlayer] == 1)
-      return 'Science';
-    if(places[currentPlayer] == 5)
-      return 'Science';
-    if(places[currentPlayer] == 9)
-      return 'Science';
-    if(places[currentPlayer] == 2)
-      return 'Sports';
-    if(places[currentPlayer] == 6)
-      return 'Sports';
-    if(places[currentPlayer] == 10)
-      return 'Sports';
-    return 'Rock';
+  const categoryLookup = {
+    0: "Pop",
+    1: "Science",
+    2: "Sports",
+    3: "Rock",
+    4: "Pop",
+    5: "Science",
+    6: "Sports",
+    7: "Rock",
+    8: "Pop",
+    9: "Science",
+    10: "Sports",
+    11: "Rock"
   };
 
-  this.createRockQuestion = function(index){
-    return "Rock Question "+index;
+  var currentCategory = () => {
+    const playerPlace = places[currentPlayer];
+    return categoryLookup[playerPlace] || "Rock";
```

Comment:
Wasn't sure how necessary `|| "Rock"` was, considering I think I sorted this problem out with the above object.